components/layout/Header.tsx

```diff
@@ -1,7 +1,22 @@
+"use client"; // Add this line to indicate a Client Component
+
 import Image from "next/image";
 import ThemeSwitch from "./ThemeSwitch";
+import React from "react";
+import { useRouter } from "next/navigation";
+
+const loginPageURL = "/signin";
```

Comment:
for constants like this some developers opt for all caps variable naming
```suggestion
const LOGIN_PAGE_URL= "/signin";
```

Another idea might be to make it an enum
```suggestion
enum Pages = {
    LOGIN_PAGE_URL = "/signin";
    ...
```